src/_zkapauthorizer/__init__.py

```diff
@@ -18,6 +18,12 @@
     "NAME",
 ]
 
+
+# Hotfix Tahoe-LAFS #3883
+from allmydata import stats
+
+stats.eventually = lambda f: f()
```

Comment:
This is the actual fix.  The rest is perhaps nice cleanup but ultimately irrelevant to the specific problem described by the related issue.

Reply:
Shouldn't it forward args + kwargs? Or does stats just happen to not ever use those...?

(It seems ... surprising .. that this fixes a memory/resource leak)

Reply:
For strict compatibility you're right - but, indeed, allmydata.stats doesn't pass any arguments so it won't make a practical difference (and I put up a Tahoe-LAFS PR to stop using `eventually` here entirely).

Regarding surprisingness ... I can say I was dismayed but not surprised that investigations led here.  Basically, `eventually` is like `reactor.callLater(0, ...)` and `reactor.callLater` takes a reference to its arguments and holds them until they run ... but most of the ZKAPAuthorizer test suite doesn't run the reactor so the references are held for the whole test suite run.  Thus, every `_Client` object the test suite ever creates lives for the full duration of the test suite.  That's not *so* destructive, but the suite is now large enough that the file descriptors for SQLite3 connections that are held open as a result are a problem.